src/main/java/org/opensearch/flowframework/util/EncryptorUtils.java

```diff
@@ -175,12 +192,23 @@ private Template processTemplateCredentials(Template template, Function<String,
     /**
      * Encrypts the given credential
      * @param credential the credential to encrypt
+     * @param tenantId The tenant id. If null, encrypts for the default tenant id.
      * @return the encrypted credential
      */
-    String encrypt(final String credential) {
-        initializeMasterKeyIfAbsent();
+    String encrypt(final String credential, @Nullable String tenantId) {
+        CountDownLatch latch = new CountDownLatch(1);
+        initializeMasterKeyIfAbsent(tenantId).whenComplete((v, throwable) -> latch.countDown());
+        try {
+            if (!latch.await(5, TimeUnit.SECONDS)) {
```

Comment:
can be moved to a constant

Reply:
the number 5?

Reply:
thought it was used all over, I see it was just 2 uses, can be ignored